### ISSUE TITLE
Room messages not received on reconnection after app swipes out 

### ISSUE THREAD

**Issue by nakkore** [processone/ejabberd#3919]
## Environment

- ejabberd version: 22.05 or 21.12
- Erlang version: Erlang (SMP,ASYNC_THREADS,HIPE) (BEAM) emulator version 10.6.4
- OS: Linux (Ubuntu)
- Installed from: source

## Bug description

Pre-condition:

Alice is connected and subscribed to room X
Bob is connected and subscribed to room X
Alice and Bob can successfully exchange room messages on their mobile devices (Android)

Scenario:

**1. Bob swipes out its app**

ejabberd realises client's connection dropped:

_2022-10-20 13:26:42.913919+00:00 [debug] Running hook c2s_closed: mod_stream_mgmt:c2s_closed/2
2022-10-20 13:26:42.914344+00:00 [info] (tls|<0.595.0>) Closing c2s connection for 001@192.168.1.28/MDAxQHRlc3Q=: Connection failed: connection closed; waiting 30 seconds for stream resumption_

**2. Alice sends a message to room X**

**3. Bob re-connects to the server WITHIN the resumption timeout**

ejabberd logs:
_2022-10-20 13:33:27.846812+00:00 [info] (<0.585.0>) Accepted connection hidden_by_ejabberd -> [::ffff:182.3.2.17]:5222
2022-10-20 13:33:27.894433+00:00 [info] (tls|<0.585.0>) Accepted c2s PLAIN authentication for 001@192.168.1.28 by external backend from hidden_by_ejabberd
2022-10-20 13:33:27.915918+00:00 [info] (tls|<0.585.0>) Opened c2s session for 001@192.168.1.28/MDAxQHRlc3Q=
2022-10-20 13:33:27.919572+00:00 [info] (unknown) Closing c2s session for 001@192.168.1.28/MDAxQHRlc3Q=: Stream closed by local host: Replaced by new connection (conflict)_

Problem:

**4. Bob doesn't receive the message sent by Alice in step 2** 

Expected behaviour:

 **Bob receives the message sent by Alice**

**Note that:**

- if Bob after step 4, disconnects and reconnects to the server, he WILL receive the expected message.
- if Alice send the message AFTER the resumption timeout, Bob will receive the message at the first re-connection in step 3
- the issue doesn't seem to be reproducible with ejabberd 19.02
- the issue doesn't seem to be reproducible if client disables stream resumption

Have you ever experienced something similar ?

Thanks
Nakkore

**Comment by licaon-kter** [comment]
Which client exactly?

Was the phone/app setup as https://dontkillmyapp.com advises?

Do put your config on https://gist.github.com

**Comment by nakkore** [comment]
@licaon-kter the client is not important here, it can suddenly disappear whenever and how it wants.
The problem that I see is that although the server knows that the client connection dropped, although the server has received Alice's message which is visible to other room participants indeed, despite these facts, Bob when reconnects doesn't receive the message at the first connection but it needs to disconnect and re-connect to receive it and when I say he didn't receive the message he didn't physically receive the message according to the packets dump.
I guess only the ejabberd devs could have a clue on what's going on.

**Comment by Neustradamus** [comment]
@nakkore: Have you tested with other XMPP clients?

Psi, Psi+, Gajim, Dino, Conversations, Monal IM, ...

**Comment by AftabUfaq** [comment]
having this issue 2025-02-05 06:49:43.436447+00:00 [info] <0.399.0>@ejabberd_listener:accept/7:344 (<0.698.0>) Accepted connection 154.198.93.105:51994 -> 172.31.87.243:5222
2025-02-05 06:49:43.437661+00:00 [debug] <0.698.0>@ejabberd_hooks:safe_apply/4:315 Running hook c2s_closed: mod_stream_mgmt:c2s_closed/2
2025-02-05 06:49:43.437746+00:00 [debug] <0.698.0>@ejabberd_hooks:safe_apply/4:315 Running hook c2s_closed: ejabberd_c2s:process_closed/2
2025-02-05 06:49:43.437827+00:00 [debug] <0.698.0>@ejabberd_hooks:safe_apply/4:315 Running hook c2s_terminated: mod_stream_mgmt:c2s_terminated/2
2025-02-05 06:49:43.437970+00:00 [debug] <0.698.0>@ejabberd_hooks:safe_apply/4:315 Running hook c2s_terminated: mod_pubsub:on_user_offline/2
2025-02-05 06:49:43.438170+00:00 [debug] <0.698.0>@ejabberd_hooks:safe_apply/4:315 Running hook c2s_terminated: ejabberd_c2s:process_terminated/2

**Comment by licaon-kter** [comment]
Which client? All clients?

Make sure you read: https://github.com/monal-im/Monal/wiki/Considerations-for-XMPP-server-admins

**Comment by AftabUfaq** [comment]
> Which client? All clients?
> 
> Make sure you read: https://github.com/monal-im/Monal/wiki/Considerations-for-XMPP-server-admins

yes, i have tried severval client apps. connverstions ans atalk, connectprp . allnhhave same issue

**Comment by AftabUfaq** [comment]
@licaon-kter can it be related to this 



#### I get 'Stream opening error'. What does that mean?

In most cases this error is caused by ejabberd advertising support for TLSv1.3 but not properly supporting it. This can happen if the OpenSSL version on the server already supports TLSv1.3 but the fast\_tls wrapper library used by ejabberd not (properly) support it. Upgrading fast\_tls and ejabberd or - theoretically - downgrading OpenSSL should fix the issue. A work around is to explicitly disable TLSv1.3 support in the ejabberd configuration. More information can be found on [this issue on the ejabberd issue tracker](https://github.com/processone/ejabberd/issues/2614).

**Comment by licaon-kter** [comment]
Keep posting in the other issue only then
